containers/layout/Layout.tsx

```diff
@@ -2,39 +2,82 @@ import React from 'react';
 import Container from '@material-ui/core/Container';
 import { I18n, TFunction } from 'next-i18next';
 import Head from 'next/head';
-import { getTitleString } from 'lib/utils/strings';
 import Footer from 'components/footer/Footer';
 import useStyles from './LayoutStyles';
 import { Box, Typography } from '@material-ui/core';
 import OptionToggler from 'components/option-toggler/OptionToggler';
 import { Button } from 'components';
 import { useAuth } from 'hooks/useAuth';
 import { useRouter } from 'next/router';
-import { FirebaseError } from 'lib/types/auth';
 import Error from 'components/error';
+import { withTranslation } from 'i18n.config';
+import { layoutError } from 'apollo/cache';
+import { useReactiveVar } from '@apollo/client';
+import { getTitleString, getTitleStringFromPathname } from 'lib/utils/strings';
+import { gql, useQuery } from '@apollo/client';
+import { getShortName } from 'lib/utils/strings';
+
+const GET_TALENT_NAME = gql`
+  query GetTalentById($id: String!) {
+    getTalentById(id: $id) {
+      id
+      basicInfo {
+        id
+        name {
+          firstName
+          middleName
+          lastName
+        }
+      }
+    }
+  }
+`;
 
 export interface LayoutProps {
-  home?: boolean;
-  heading?: string;
-  title?: string | string[];
-  error: FirebaseError | null;
   children?: React.ReactNode;
   i18n?: I18n;
   t: TFunction;
 }
 
-const Layout = ({
-  home = false,
-  error = null,
-  children,
-  heading,
-  title,
-  i18n,
-  t,
-}: LayoutProps): React.ReactElement => {
+const Layout = ({ children, t, i18n }: LayoutProps): React.ReactElement => {
   const classes = useStyles();
   const auth = useAuth();
+  const reactiveError = useReactiveVar(layoutError);
   const router = useRouter();
+  const { pathname, query } = router;
+
+  const isHome = pathname === '/';
+  const loadBasicInfo = pathname.includes('/talents/[id]'); // dependent on how profile page is named
+
+  const { id } = query;
+
+  const { data, error } = useQuery(GET_TALENT_NAME, {
+    variables: {
+      id,
+    },
+    skip: !loadBasicInfo,
+  });
+
+  if (error) {
```

Comment:
do we really want to insert the 404 page in Layout?

Reply:
Good point. Let's not insert the error. If something goes wrong, i.e. there is an `error` then there should be no `data` either. If there are no `data`, then the title will fall back to this `else` statement:

```
else {
    title = getTitleStringFromPathname(pathname);
  }
```

So I just took the `error` handling out.